bcgov/GDX-Analytics-microservice
# Gdxdsd 7484 sdpr 20min feed

### This PR does the following
Creates a new job based off the existing sdpr_hourly files that will run in 20 min intervals.
Please note that some files were modified for testing purposes. You can see these modified files attached in the ticket.

### Login to development environment

1. Run the following command to login to EC2
```
$ awsmfa prod ######
Generating new IAM STS Token ...
STS Session Token generated and saved in profile mfa successfully.
$ microservice_ssm
```
2. Run the following command to change to the development branch:
```
$ cd /home/microservice/branch/GDXDSD-7484_sdpr_20min_feed/redshift_to_s3
```

### Review files

1. Review [sdpr_20min.json](https://github.com/bcgov/GDX-Analytics-microservice/blob/GDXDSD-7484_sdpr_20min_feed/redshift_to_s3/config.d/sdpr_20min.json)
2. Review the test file GDXDSD-7484_sdpr_20min.json and compare it with [sdpr_20min.json](https://github.com/bcgov/GDX-Analytics-microservice/blob/GDXDSD-7484_sdpr_20min_feed/redshift_to_s3/config.d/sdpr_20min.json). Make sure the only change is to `directory` 
3. Review [sdpr_20min.sql](https://github.com/bcgov/GDX-Analytics-microservice/blob/GDXDSD-7484_sdpr_20min_feed/redshift_to_s3/dml/sdpr_20min.sql). Specifically the `WHERE` clause to make sure it grabs data from the start of current day up to the time the job runs

### Testing instructions:

1. Run the following command to test the new job, and compare to the expected output:
```
$ /home/microservice/.local/bin/pipenv run python redshift_to_s3.py -c config.d/GDXDSD-7484_sdpr_20min.json

***The microservice ran successfully***

Report: redshift_to_s3.py

Config: config.d/GDXDSD-7484_sdpr_20min.json

DML: sdpr_20min.sql

Microservice started at: 2024-11-27 09:52:21-0800 (PST), ended at: 2024-11-27 09:52:25-0800 (PST), elapsing: 0:00:03.996948.


Objects loaded to S3 /batch: 1/1
Objects successfully loaded to S3 /batch: 1

List of objects successfully loaded to S3 /batch
1. processed/batch/client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/sdpr_20min_20241127T175221


Objects to store: 1
Objects stored to s3 /client: 1

List of objects stored to S3 /client:
1: client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/sdpr_20min_20241127T175221_part000.csv


Objects to process: 1
Objects processed to s3 /good: 1

List of objects processed to S3 /good:
1: processed/good/client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/sdpr_20min_20241127T175221_part000
$ 
```
2. Check to make sure the file specified in the console output appears in the s3 batch folder: https://ca-central-1.console.aws.amazon.com/s3/buckets/sp-ca-bc-gov-131565110619-12-microservices?region=ca-central-1&bucketType=general&prefix=processed/batch/client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/&showversions=false
3. Check to make sure the file specified in the console output appears in the s3 client folder: https://ca-central-1.console.aws.amazon.com/s3/buckets/sp-ca-bc-gov-131565110619-12-microservices?prefix=client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/&region=ca-central-1&bucketType=general
4. Check to make sure the file specified in the console output appears in the s3 good folder: https://ca-central-1.console.aws.amazon.com/s3/buckets/sp-ca-bc-gov-131565110619-12-microservices?region=ca-central-1&bucketType=general&prefix=processed/good/client/doug_test/GDXDSD-7484/theq_sdprabi/sdpr_20min/v01-Nov_2024_20min_feed/&showversions=false
